File: text-processing/brute-force.js
function find_brute(sample, pattern) {
    let sampleLen = sample.length, 
        patternLen = pattern.length,
        endPoint = sampleLen - patternLen + 1

    for (let i = 0; i < endPoint; i++) {
        overlapSize = 0
        
        while (overlapSize < patternLen && sample[i + overlapSize] === pattern[overlapSize]) {
            overlapSize += 1
        }
            
        if (overlapSize === patternLen) {
            return i
        }
    }
       
    return -1
}
 

File: text-processing/brute-force.py
def find_brute(sample, pattern):
    sample_len, pattern_len = len(sample), len(pattern)

    for i in range(sample_len - pattern_len + 1):
        overlap_size = 0

        while overlap_size < pattern_len and sample[i + overlap_size] == pattern[overlap_size]:
            overlap_size += 1

        if overlap_size == pattern_len:
            return i
    return -1